packages/shared/src/messages/MessageProxy.ts

```diff
@@ -52,9 +51,16 @@ export function isMessageResponse(content: unknown): content is IMessageResponse
   return isMessageRequest(content) && 'status' in content;
 }
 
+// instance has methods that are callable
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type ObjectInstance<T> = {
+  [key: string]: (...args: unknown[]) => unknown;
```

Comment:
```suggestion
  [key: string]: (...args: unknown[]) => Promise<unknown>;
```

Can we force Promise return type?

Reply:
done